### DISCUSSION TITLE
What's the difference between Tekton Pipelines and OpenShift Pipelines?

### DISCUSSION THREAD

**Discussion by worldofgeese** [tektoncd/pipeline#7323] [Q&A] [answered]
I have a SNO OpenShift cluster I use. Today, I made the decision to install Shipwright (which requires Tekton) on the cluster. So I installed Tekton then Shipwright.

- Could I have installed OpenShift Pipelines instead?
- What, if any, differences are there from a user perspective?

**Comment by khrm** [comment, marked as answer]
1. Red Hat Provides support for OpenShift Pipelines.
2. OpenShift Pipelines leverages Operator to install Pipelines. Operators also create a `serviceAccount` `pipelines`  using an RBAC controller. Otherwise, you need to create these on your own if you like to build images and such. It also installs other components like Triggers, PAC, and Chains. You can use the operator to install supported Results and Hub.

**Reply by worldofgeese** [reply]
Is it fair to say Tekton is the upstream for OpenShift Pipelines in the same way OKD is upstream of OpenShift or Shipwright is upstream of what will eventually become OpenShift builds version 2?

**Reply by khrm** [reply]
Yes. That's correct.

**Reply by titou10titou10** [reply]
yes shipwright will become ocp build v2

https://developers.redhat.com/articles/2021/06/17/shipwright-framework-building-container-images-kubernetes#

> Shipwright already powers IBM Cloud Code Engine's build system and will form the basis for Red Hat OpenShift builds version 2, which is expected to be launched in technical preview later this year.
